src/stirling/source_connectors/socket_tracer/protocols/mqtt/parse_test.cc

```diff
@@ -0,0 +1,914 @@
+            // message identifier
+            0x00, 0x01};
+    uint8_t kPubcompFrame[] = {
+            // header flags
+            0x70,
+            // message length
+            0x02,
+            // message identifier
+            0x00, 0x01};
+    uint8_t kSubscribeFrame[] = {
+            // header fields
+            0x82,
+            // message length
+            0x10,
+            // message identifier
+            0x00, 0x01,
+            // properties length
+            0x00,
+            // topic length
+            0x00, 0x0a,
+            // topic
+            0x74, 0x65, 0x73, 0x74, 0x2f, 0x74, 0x6f,
+            0x70, 0x69, 0x63,
+            // subscription options
+            0x00};
+    uint8_t kSubackFrame[] = {
+            // header flags
+            0x90,
+            // message length
+            0x04,
+            // message identifier
+            0x00, 0x01,
+            // properties length
+            0x00,
+            // reason code
+            0x00};
+    uint8_t kUnsubscribeFrame[] = {
+            // header flags
+            0xa2,
+            // message length
+            0x0f,
+            // message identifier
+            0x00, 0x02,
+            // properties length
+            0x00,
+            // topic length
+            0x00, 0x0a,
+            // topic
+            0x74, 0x65, 0x73, 0x74, 0x2f, 0x74, 0x6f,
+            0x70, 0x69, 0x63};
+    uint8_t kUnsubackFrame[] = {
+            // header flags
+            0xb0,
+            // message length
+            0x04,
+            // message identifier
+            0x00, 0x02,
+            // properties length
+            0x00,
+            // reason code
+            0x00};
+    uint8_t kPingreqFrame[] = {
+            // header flags
+            0xc0,
+            // message length
+            0x00};
+    uint8_t kPingrespFrame[] = {
+            // header flags
+            0xd0,
+            // message length
+            0x00};
+    uint8_t kDisconnectFrame[] = {
+            // header flags
+            0xe0,
+            // message length
+            0x01,
+            // reason code
+            0x04};
+
+    frame_view = CreateStringView<char>(CharArrayStringView<uint8_t>(kConnectFrame));
+    result_state = ParseFrame(message_type_t::kRequest, &frame_view, &frame);
+    ASSERT_EQ(result_state, ParseState::kSuccess);
+    EXPECT_EQ(frame.control_packet_type, "CONNECT");
+    EXPECT_EQ(frame.header_fields["remaining_length"], (size_t) 16);
+    EXPECT_EQ(frame.header_fields["username_flag"], (unsigned long) 0);
+    EXPECT_EQ(frame.header_fields["password_flag"], (unsigned long) 0);
+    EXPECT_EQ(frame.header_fields["will_retain"], (unsigned long) 0);
+    EXPECT_EQ(frame.header_fields["will_qos"], (unsigned long) 0);
+    EXPECT_EQ(frame.header_fields["will_flag"], (unsigned long) 0);
+    EXPECT_EQ(frame.header_fields["clean_start"], (unsigned long) 1);
+    EXPECT_EQ(frame.header_fields["keep_alive"], (unsigned long) 60);
```

Comment:
Please use `UL` to form the appropriate literals or remove the casting entirely. The tests seem to pass without these, so my preference is the latter. This applies to the other casting done in this file.